lib/src/block_syntaxes/unordered_list_syntax.dart

```diff
@@ -10,6 +11,21 @@ class UnorderedListSyntax extends ListSyntax {
   @override
   RegExp get pattern => ulPattern;
 
+  @override
+  bool canParse(BlockParser parser) {
+    // Check if it matches `hrPattern`, otherwise it will produce an infinite
```

Comment:
Thanks for catching this; I experienced this but didn't figure out what was causing it.